src/meshlink.c

```diff
@@ -2269,19 +1966,22 @@ bool meshlink_set_port(meshlink_handle_t *mesh, int port) {
 		goto done;
 	}
 
-	close_network_connections(mesh);
-	exit_configuration(&mesh->config);
-
-	char portstr[10];
-	snprintf(portstr, sizeof(portstr), "%d", port);
-	portstr[sizeof(portstr) - 1] = 0;
+	/* Write meshlink.conf with the updated port number */
+	write_main_config_files(mesh);
 
-	modify_config_file(mesh, mesh->name, "Port", portstr, true);
+	/* Close down the network. This also deletes mesh->self. */
+	close_network_connections(mesh);
 
```

Comment:
Storing the new port number into the mesh handle is missing here.

Reply:
Storing the new port number is done by the call to `write_main_config_files()` right above these lines. But indeed, mesh->myport still needs to be set correctly before calling that.